lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -203,6 +203,7 @@ public void close()
         try (outputStream) {
             columnWriters.forEach(ColumnWriter::close);
             flush();
+            columnWriters = ImmutableList.of();
```

Comment:
is this related to the memory accounting?

Reply:
Yes, now that we are accounting for buffered pages I wanted to ensure that we release them and reduce retained memory as soon as possible. Buffered pages have to be held until they are flushed to output stream, so this is the most straightforward way to release them.
I think this memory would also get released by the current code in connector page sink calling commit on writer and then releasing reference to it. But that is a few layers above and would delay the cleanup a bit.

Reply:
ok, any reason to use an empty list instead of `null` (it starts as `null`)?

Reply:
constructor calls `initColumnWriters` which ensures that it is never null, there are many usages of this field without a null check